lib/__tests__/init.test.ts

```diff
@@ -259,4 +259,40 @@ describe("init", () => {
       });
     });
   });
+
+  describe("userToken param", () => {
+    let setUserToken;
+    let setAnonymousUserToken;
+    beforeEach(() => {
+      setUserToken = jest.spyOn(analyticsInstance, "setUserToken");
+      setAnonymousUserToken = jest.spyOn(
+        analyticsInstance,
+        "setAnonymousUserToken"
+      );
+    });
+
+    afterEach(() => {
+      setUserToken.mockRestore();
+      setAnonymousUserToken.mockRestore();
+    });
+
+    it("should set userToken", () => {
+      analyticsInstance.init({ apiKey: "***", appId: "XXX", userToken: "abc" });
+      expect(setUserToken).toHaveBeenCalledTimes(1);
+      expect(setUserToken).toHaveBeenCalledWith("abc");
+    });
+
+    it("shouldn't set anonymous user token to cookie", () => {
+      analyticsInstance.init({
+        apiKey: "***",
+        appId: "XXX",
+        userToken: "abc",
+        useCookie: true
+      });
+      expect(setUserToken).toHaveBeenCalledTimes(1);
+      expect(setUserToken).toHaveBeenCalledWith("abc");
```

Comment:
can we add a test with a Manual setUserToken afterwards?

Reply:
https://github.com/algolia/search-insights.js/pull/241/commits/6aa27679a31a1aea86b7f9a5ad8f4d034663d963